backend/RoomPlanner/room-planner/src/main/java/edu/roomplanner/service/impl/UserServiceImpl.java

```diff
@@ -12,20 +18,35 @@
 public class UserServiceImpl implements UserService {
 
     private UserRepository userRepository;
+    private UserValidation userValidation;
+    private RoomDtoMapper roomDtoMapper;
 
     @Autowired
-    public UserServiceImpl(UserRepository userRepository) {
+    public UserServiceImpl(UserRepository userRepository, UserValidation userValidation, RoomDtoMapper roomDtoMapper) {
         this.userRepository = userRepository;
+        this.userValidation = userValidation;
+        this.roomDtoMapper = roomDtoMapper;
+    }
+
+    @Override
+    public List<RoomDto> getAllRooms() {
+        List<UserEntity> roomEntities = userRepository.findByType(UserType.ROOM);
+        return roomDtoMapper.mapEntityListToDtoList(roomEntities);
     }
 
     @Override
-    public List<UserEntity> getUsers() {
-        return (List<UserEntity>) userRepository.findAll();
+    public RoomDto getRoomById(Long id) {
+        RoomDto roomDto = null;
+        if (userValidation.checkValidRoomId(id)) {
+            UserEntity userEntity = userRepository.findById(id).get();
+            roomDto = roomDtoMapper.mapEntityToDto((RoomEntity) userEntity);
+        }
+        return roomDto;
     }
 
     @Override
-    public UserEntity saveUser(UserEntity userEntity) {
-        return userRepository.save(userEntity);
+    public List<RoomDto> getFilteredRooms() {
```

Comment:
Why did you add this method here?
The filtering can be included in get all rooms.